src/main/scala/quantiles/QuantileNonSample.scala

```diff
@@ -0,0 +1,252 @@
+package quantiles
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+import scala.util.control.Breaks._
+import scala.util.Sorting.quickSort
+
+class QuantileNonSample[T](val sketchSize: Int,
+                           private val shrinkingFactor: Double = 0.64)
+                          (implicit ordering: Ordering[T],
+                           ct: ClassTag[T]) extends Serializable{
+
+  private var curNumOfCompactors = 0
+  // initialize with ArrayBuffer, add compactors later
+  private var compactors = ArrayBuffer[NonSampleCompactor[T]]()
+  // number of items in compactors
+  private var compactorActualSize = 0
+  // overall capacity of compactors
+  private var compactorTotalSize = 0
+  expand()
+
+
+
+  // expand a layer of compactor
+  private def expand(): Unit = {
+    compactors = compactors :+ new NonSampleCompactor[T]
+    curNumOfCompactors = compactors.length
+    var size = 0
+    (0 until curNumOfCompactors).foreach { height =>
+      size = size + capacity(height)
+    }
+    compactorTotalSize = size
+  }
+
+  private def capacity(height:Int): Int = {
+    Math.ceil(sketchSize * Math.pow(shrinkingFactor, height)).toInt + 1
```

Comment:
Capacity should be even, and at least 2
Can change to 
2 * (Math.ceil(sketchSize * Math.pow(shrinkingFactor, height) / 2).toInt + 1)

Reply:
Actually Math.ceil(x) function ensures that as long as x is positive then the Math.ceil(sketchSize * Math.pow(shrinkingFactor, height)).toInt + 1
should be at least 2, but it didn't take care of the even capacity requirement so I will happily accept the change.